### PR TITLE
Create manual-check-external-pr.yml

### PR DESCRIPTION
**Pull request by psaavedra**
This Github Action allows to manually create a new branch in the CI mirror repo the under the `pull/` path based on the changes suggested by an external Pull Request from an forked repository. This push triggers the CI jobs in the Gitlab CI.

This action **does**:

* force the push of the `pull/<id>/HEAD` branch in the mirror. It also notifies to the Gitlab CI for a immediate pull of the mirror (this last is not strictly necessary)
* Puts a comment into the Pull Request pointing to the created branch

This action has some **limitations**:

* ~~The results of the CI jobs are not linked to the original Pull Request. Instead of that that relies in a manual action by the reviewer who have to put a comment in the PR linking the results.~~. I've noticed that this works if the references (hash) for the HEAD commit what triggers the pipeline is the same that the `HEAD` in the `pull/<id>/HEAD` branch.
* This action has to be manually triggered by a reviewer. To run code from forked repos still implies several security concerns . See: https://gitlab.com/gitlab-org/gitlab/-/issues/5667#permissions-and-security

### CODE REVIEW

File: .github/workflows/manual-check-external-pr.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+       chmod 700 ~/.ssh
+       ssh-keyscan -t rsa gitlab.com github.com >>~/.ssh/known_hosts
+       git config --global user.name "${{ env.BOT_USER }}"
+       git config --global user.email "${{ env.BOT_EMAIL }}"
+       git config --global pull.rebase "preserve"
+       git clone ${{ env.REPO }} repo
+       cd repo
+       git fetch ${{ env.REPO }} pull/${{ github.event.inputs.pr-id }}/head:check/pr_${{ github.event.inputs.pr-id }}_build
+       git checkout check/pr_${{ github.event.inputs.pr-id }}_build
+       git push --force origin check/pr_${{ github.event.inputs.pr-id }}_build
```
Comment (philn) [inline .github/workflows/manual-check-external-pr.yml:36]: Why is this needed? Is the gitlab thing checking this ref? Maybe instead it can check the existing ref that corresponds to the PR?
Also, when is this ref cleaned up?

Comment (psaavedra) [reply .github/workflows/manual-check-external-pr.yml]: Added  a manual action to clean up these refs in the mirror repository. We don't have a huge amount of PR activity and still to keep around the PR references in the mirror repo for a while it can be still convenient  so the clean up process can be triggered manually every then and now. 
Change: 7f3b70c2947386f64986603c7ce16c974045304b

Comment (psaavedra) [reply .github/workflows/manual-check-external-pr.yml]: > Why is this needed? Is the gitlab thing checking this ref? Maybe instead it can check the existing ref that corresponds to the PR?

Gitlab Mirror is not pulling the _special_ `/pull/*` branches from Github. That is inhibiting the `push` notifications for the `pull_requests` for the Gitlab CI webhooks integration. 

This solution is based on the original idea published [here](https://gitlab.com/gitlab-org/gitlab/-/issues/5667#note_144850858) but with slightly differences:

* The logic is encapsulated in a Github action not in a `special` (check_pr) branch in the mirror
* The trigger is the **Start** button for this **Github action**

More literature here:  https://gitlab.com/gitlab-org/gitlab/-/issues/5667

**Review by philn — APPROVED** [review]
